presto-main/src/main/java/com/facebook/presto/security/AccessControlManager.java

```diff
@@ -157,11 +160,21 @@ public void checkCanSetUser(Principal principal, String userName)
     }
 
     @Override
+    public void checkCanAccessCatalog(Identity identity, String catalogName)
+    {
+        if (filterCatalogs(identity, ImmutableSet.of(catalogName)).isEmpty()) {
+            denyCatalogAccess(catalogName);
+        }
+    }
+
+    @Override
     public void checkCanCreateSchema(TransactionId transactionId, Identity identity, CatalogSchemaName schemaName)
     {
         requireNonNull(identity, "identity is null");
         requireNonNull(schemaName, "schemaName is null");
 
+        authenticationCheck(() -> checkCanAccessCatalog(identity, schemaName.getCatalogName()));
```

Comment:
Could you check catalog access when you get the catalog handle? e.g. MetadataManager.getCatalogHandle?